src/mount.c

```diff
@@ -104,10 +184,17 @@ pu_umount_all(const gchar *device,
 gboolean
 pu_device_mounted(const gchar *device)
 {
-    g_autofree gchar *cmd = g_strdup("mount");
-    g_autofree gchar *output = NULL;
+    g_autoptr(GList) mounted = NULL;
 
-    g_spawn_command_line_sync(cmd, &output, NULL, NULL, NULL);
+    g_return_val_if_fail(g_strcmp0(device, "") > 0, FALSE);
+
+    if (!pu_find_mounted(device, &mounted)) {
+        g_printerr("Failed checking if device is mounted");
+        return TRUE;
```

Comment:
We really should return `FALSE` here instead. Returning `TRUE` creates the impression that `pu_device_mounted()` successfully found a mounted device, while in reality this is not the case. Then, introducing GError seems like a good idea as well.